include/miral/miral/runner.h

```diff
@@ -90,6 +105,14 @@ class MirRunner
     MirRunner& operator=(MirRunner const&) = delete;
     struct Self;
     std::unique_ptr<Self> const self;
+
+    struct SignalInfo
+    {
+        std::initializer_list<int> signals;
+        std::function<void(int)> const& handler;
+    };
+
+    void unregister_fd_handler(void const* owner);
```

Comment:
Why are we declaring a private function in a public header? Users cannot call it and we don't. If we did use it, any functionality would reside on `Self` and should be invoked directly.